### PR TITLE
[SPARK-6313] Add config option to disable file locks/fetchFile cache to ...

### PR DESCRIPTION
**Pull request by nemccarthy**
...support NFS mounts.

This is a work around for now with the goal to find a more permanent solution.
https://issues.apache.org/jira/browse/SPARK-6313

### CODE REVIEW

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test.

**Comment by shaneknapp** [comment]
jenkins, test this please

**Comment by JoshRosen** [comment]
As a workaround for the NFS issue, this looks good to me. I have a couple of minor grammar / documentation nits, which I'll comment on below, but I can fix those up myself on merge if you don't get around to it first.

File: docs/configuration.md [hunk 1]
```diff
@@ -746,6 +746,18 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
+  <td><code>spark.files.useFetchCache</code></td>
+  <td>true</td>
+  <td>
+    If file fetching should use local caching. The improves performance when running multiple
```
Comment (JoshRosen) [inline docs/configuration.md]: I'd reword this slightly in order to match the style of similar boolean options.  How about this:

> If set to true (default), file fetching will use a local cache that is shared by executors that belong to the same application, which can improve task launching performance when running many executors on the same host.  If set to false, these caching optimizations will be disabled and all executors will fetch their own copies of files.  This optimization may be disabled in order to use NFS filesystems as Spark local directories (see <a href="https://issues.apache.org/jira/browse/SPARK-6313">SPARK-6313</a> for more details).

**Comment by nemccarthy** [comment]
@JoshRosen Thanks for the feedback. Updated as suggested.

**Comment by JoshRosen** [comment]
LGTM, so I'm going to merge this into `master` (1.4.0), `branch-1.3` (1.3.1), and `branch-1.2` (1.2.2) (since this cache was first introduced in 1.2.0).
